### PR TITLE
Fix enum search select

### PR DESCRIPTION
**Pull request by colin27091**

### CODE REVIEW

File: packages/next-admin/src/utils/props.ts [hunk 1]
```diff
@@ -154,15 +155,17 @@ export async function getPropsFromParams({
       const edit = options?.model?.[resource]?.edit as EditOptions<
         typeof resource
       >;
-      schema = transformSchema(schema, resource, edit);
-      schema = await fillRelationInSchema(
-        schema,
+      
+      let deepCopySchema = cloneDeep(schema);
```
Comment (foyarash) [inline packages/next-admin/src/utils/props.ts:159]: Why are we doing this ?

**Review by foyarash — COMMENTED** [review]
I wonder if we can just do a client side research in case of an enum ?

**Comment by colin27091** [comment]
> I wonder if we can just do a client side research in case of an enum ?

Enum and relationship select are the same component, they both use the same search system (server side), to avoid splitting this component I have kept server side search even on enum.

File: packages/next-admin/src/utils/props.ts (hunk 1 above)
Comment (colin27091) [reply packages/next-admin/src/utils/props.ts:159]: During some operations we make some changes to `schema`, if these changes are made to original `schema` they will persist for next operations. By cloning `schema` we can preserve the original `schema`.

**Pull request merged**